### PR TITLE
#3 StreamTask

### PR DESCRIPTION
**Pull request by KluevJakov**

### CODE REVIEW

File: src/test/java/com/github/KluevJakov/myinternship/StreamTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+
+public class StreamTest {
+    @Test
+    public void negativeCheck() {
+        Map<Integer, List<String>> check = new HashMap<>();
+
+        check.put(1,new ArrayList<>(Arrays.asList("o5")));
```
Comment (kovinz) [inline src/test/java/com/github/KluevJakov/myinternship/StreamTest.java:15]: add space after comma please

Comment (kovinz) [reply src/test/java/com/github/KluevJakov/myinternship/StreamTest.java:15]: the same applies to other tests also

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StreamTask {
+    public static Map<Integer, List<String>> groupByDigitNumbers(int[] input) {
+
+        Map<Integer, List<String>> result = Arrays.stream(input)
+                .boxed()
```
Comment (kovinz) [inline src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:13]: remove this step - you box int to Integer on this step and on the next "filter" step when you compare it to 0 you will unbox every item then compare it and then box it again - seems like too much work for nothing

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                .filter(e -> e > 0)
```
Comment (kovinz) [inline src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:14]: vars in lambda should also have some meaningful name - please rename

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+                .sorted()
```
Comment (kovinz) [inline src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:15]: why do you need sorting in this task?

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java (hunk 3 above)
Comment (kovinz) [reply src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:14]: also task says "filter out negative values". Zero isn't negative

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+                .map(e -> e % 2 == 0 ? "e" + e : "o" + e)
```
Comment (kovinz) [inline src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:16]: please rename var in lambda

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+                .collect(Collectors.toMap(
```
Comment (kovinz) [inline src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:17]: Collectors have better method for grouping by some attribute - please check java doc for Collectors class and find it :)

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class StreamTask {
+    public static Map<Integer, List<String>> groupByDigitNumbers(int[] input) {
+
+        Map<Integer, List<String>> result = Arrays.stream(input)
```
Comment (kovinz) [inline src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:12]: why not to remove variable "result" and just return result of stream?
P.S. IDEA proposes the same and shows warning

File: src/test/java/com/github/KluevJakov/myinternship/StreamTest.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
+package com.github.KluevJakov.myinternship;
+
+import com.github.KluevJakov.myinternship.tasks.StreamTask;
+import org.junit.Test;
+
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+
+public class StreamTest {
```
Comment (kovinz) [inline src/test/java/com/github/KluevJakov/myinternship/StreamTest.java:10]: please add tests for negative cases - such as "all values are negative", "empty array is passed to the method"

Comment (kovinz) [reply src/test/java/com/github/KluevJakov/myinternship/StreamTest.java:10]: plus after fixing bug with 0 - add test where 0 will be present among other items
this is a good practice to add UT to check for a fixed bug

**Review by kovinz — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java (hunk 4 above)
Comment (KluevJakov) [reply src/main/java/com/github/KluevJakov/myinternship/tasks/StreamTask.java:14]: From the example, it seemed to me that the numbers should also be sorted in ascending order, but after rereading, I realized that the assignment did not say about this

**Review by kovinz — APPROVED** [review]
